includes/functions.php

```diff
@@ -2310,4 +2374,901 @@ function tsml_date_localised($format, $timestamp = null)
+                echo 'Contact 3 Email --→ ' . $import_meeting['name'] . '<br>';
+                echo $db_meeting['contact_3_email'] . '  <br>';
+                echo $import_meeting['contact_3_email'] . ' <br><br>';
+            }
+
+            return false;
+        }
+    }
+
+    if (!empty($db_meeting['contact_3_phone']) && !empty($import_meeting['contact_3_phone'])) {
+        if ($db_meeting['contact_3_phone'] !== $import_meeting['contact_3_phone']) {
+
+            if ($test_mode) {
+                echo 'Contact 3 Phone --→ ' . $import_meeting['name'] . '<br>';
+                echo $db_meeting['contact_3_phone'] . '  <br>';
+                echo $import_meeting['contact_3_phone'] . ' <br><br>';
+            }
+
+            return false;
+        }
+    }
+
+    //do Incremental compare of only those records which have changed since the last update occurred
+
+    if (!empty($import_meeting['updated'])) {
+        //localize imported timestamp
+        $imported_timestamp = strtotime(tsml_date_localised(get_option('date_format') . ' ' . get_option('time_format'), strtotime($import_meeting['updated'])));
+
+		$current_meeting_last_update = strtotime($import_meeting['updated']);
+		//if ($current_meeting_last_update > $data_source_last_update) {
+        if ($imported_timestamp > $data_source_last_update) {
+
+            if ($test_mode) {
+                $ds_last_update = date(get_option('date_format') . ' ' . get_option('time_format'), $data_source_last_update);
+                $cur_last_update = date(get_option('date_format') . ' ' . get_option('time_format'), $imported_timestamp);
+
+                echo 'Updated --→ ' . $import_meeting['name'] . '<br>';
+                echo $ds_last_update . '  <br>';
+                echo $cur_last_update . ' <br><br>';
+            }
+
+            return false;
+
+        } else {
+
+            return true;
+
+        }
+    }
+
+    //no changes, so we are in sync
+    return true;
+}
+//function to remove special characters and unwanted text from the passed string
+//used:	here (called from tsml_verify_identical_records)
+function tsml_remove_special_char($str)
+{
+
+    $str = stripslashes($str);
+    $result = str_replace(array('\'', '\’', '"', ',', '.', ';', '<', '>', '&', '"', "'", "’", " ", "Â", "Â’", "%c2%92", "%20", "#nbsp", "#039", "amp", "nbsp", "rsquo", "quot"), '', $str);
+
+    return $result;
+}
+
+//function to convert a class object to an array
+//used:	here (called from tsml_verify_identical_records)
+function tsml_object_to_array($data)
+{
+    if (is_array($data) || is_object($data)) {
+        $result = [];
+        foreach ($data as $key => $value) {
+            $result[$key] = (is_array($value) || is_object($value)) ? tsml_object_to_array($value) : $value;
+        }
+        return $result;
+    }
+    return $data;
+}
+
+//function to clean up filename string to make it safe
+//used:	admin_import
+function tsml_clean_file_name($file_name)
```

Comment:
this function isn't being called

Reply:
I believe your right. It was part of the project uniting the CSV code to use change detection. We will remove.